frontend/src/Products.js

```diff
@@ -0,0 +1,209 @@
+                  onClick={() => {
+                    dispatch({ type: "SORT", payload: "PRICE_HIGH_TO_LOW" });
+                  }}
+                />
+                Price high to low
+              </li>
+              <li>
+                <input
+                  type="radio"
+                  name="sort1"
+                  checked={sortBy === "PRICE_LOW_TO_HIGH"}
+                  onClick={() => {
+                    dispatch({ type: "SORT", payload: "PRICE_LOW_TO_HIGH" });
+                  }}
+                />
+                Price low to high
+              </li>
+            </ul>
+            Filter
+            <ul class="list ">
+              <li>
+                <input
+                  type="checkbox"
+                  checked={!showInventoryAll}
+                  onChange={() => dispatch({ type: "TOGGLE_INVENTORY" })}
+                />
+                Include out of stock
+              </li>
+              <li>
+                <input
+                  type="checkbox"
+                  checked={showFastDeliveryOnly}
+                  onChange={() => dispatch({ type: "TOGGLE_DELIVERY" })}
+                />
+                Fast Delivery
+              </li>
+              <li>Price Range</li>
+              <li>
+                {`Rs:0`}
+                <input
+                  id="PRICE_RANGE"
+                  name="PRICE_RANGE"
+                  type="range"
+                  min="0"
+                  max="12000"
+                  step="10"
+                  value={range}
+                  onChange={(e) =>
+                    setrange((prev) =>
+                      e.target.value < 100 ? 100 : e.target.value
+                    )
+                  }
+                />
+                {`Rs:${range}`}
+              </li>
+            </ul>
+          </div>
+          <div className="items">
+            {rangefilteredData.map(
+              ({
+                _id,
+                name,
+                image,
+                price,
+                productName,
+                inStock,
+                level,
+                fastDelivery,
+
+                offer,
+                ratings,
+                isnew
+              }) => (
+                <div
+                  class="cards-t1"
+                  style={{ width: "40%", maxWidth: "210px" }}
+                >
+                  <div class="ratewrapper">
+                    <img class="cards-t1-img" src={image} alt={name} />
+                    <div className="rate">
+                      {ratings}
+                      <i class="fa fa-star"></i>
+                    </div>
+                  </div>
+                  {isnew ? <div class="cards-badge">New</div> : ""}
+
+                  <div
+                    class="title"
+                    style={{
+                      overflow: "hidden",
+                      textOverflow: "..2.",
+                      height: "20px"
+                    }}
```

Comment:
Add a little padding, in the title of card as the big names are going till end of card

Reply:
Thanks for pointing this out. I just missed this part. I will add some padding for the title.